earthmover/nodes/source.py

```diff
@@ -266,13 +265,29 @@ def __get_skiprows(config: 'YamlMapping'):
             _header_rows = config.get('header_rows', 1)
             return int(_header_rows) - 1  # If header_rows = 1, skip none.
 
+        def __read_fwf(file: str, config: 'YamlMapping'):
+            colspec_file = config.get('colspec_file')
+            if colspec_file:
+                try:
+                    file_format = pd.read_csv(os.path.join(os.path.dirname(self.config.__file__), colspec_file))
+                # we need to handle this separately because otherwise EM will report that the source file
+                # (instead of the colspec file) could not be found
+                except FileNotFoundError:
+                    self.error_handler.throw(
+                        f"colspec file '{colspec_file}' not found"
+                    )
+                colnames = file_format.field_name
+                colspecs = list(zip(file_format.start_index, file_format.end_index))
+                return dd.read_fwf(file, colspecs=colspecs, header=config.get('header_rows', "infer"), names=colnames, converters={c:str for c in colnames})
```

Comment:
Add a couple of variables here to clean up these read lines. We should technically be using `error_handler.assert_get_key()` when retrieving variables from the YAML config blocks.

Reply:
Could you be more specific about what kind of cleanup you want to see? 

I'm happy to use `assert_get_key` although I notice that none of the other read lambdas use it, and it will make this code more verbose